shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -428,6 +446,25 @@ - (void)sendPointerEvent:(const FlutterPointerEvent&)event {
   _embedderAPI.SendPointerEvent(_engine, &event, 1);
 }
 
+- (void)updateSemanticsEnabled:(BOOL)enabled {
+  if (!enabled && _bridge) {
+    _bridge = std::shared_ptr<flutter::AccessibilityBridge>(nullptr);
+  } else if (enabled && !_bridge) {
+    _bridge = std::make_shared<flutter::AccessibilityBridge>(
+        std::make_unique<flutter::AccessibilityBridgeMacDelegate>(self));
```

Comment:
I think we probably want this to be passed as a `scoped_nsobject<FlutterEngine>([self retain])`.

But that comes from FML. I'm not sure what we do about this in the desktop embedding.

At any rate, it'd be nice to avoid raw pointer usage here. @stuartmorgan probably knows what pattern(s) are used in the desktop embedding better.